From: Jeff Mahoney <jeffm@suse.com>
Subject: [PATCH 15/40] reiserfs: remove i_has_xattr_dir

 With the changes to xattr root locking, the i_has_xattr_dir flag
 is no longer needed. This patch removes it.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>

---
 fs/reiserfs/xattr.c |    7 -------
 1 files changed, 7 deletions(-)

--- linux-2.6.24.orig/fs/reiserfs/xattr.c
+++ linux-2.6.24/fs/reiserfs/xattr.c
@@ -399,7 +399,6 @@ reiserfs_xattr_set(struct inode *inode, 
 
 	xahash = xattr_hash(buffer, buffer_size);
 	xinode = fp->f_path.dentry->d_inode;
-	REISERFS_I(inode)->i_flags |= i_has_xattr_dir;
 
 	/* we need to copy it off.. */
 	if (xinode->i_nlink > 1) {
@@ -518,7 +517,6 @@ reiserfs_xattr_get(const struct inode *i
 
 	xinode = fp->f_path.dentry->d_inode;
 	isize = xinode->i_size;
-	REISERFS_I(inode)->i_flags |= i_has_xattr_dir;
 
 	/* Just return the size needed */
 	if (buffer == NULL) {
@@ -739,9 +737,6 @@ int reiserfs_delete_xattrs(struct inode 
 	fput(fp);
 
       out:
-	if (!err)
-		REISERFS_I(inode)->i_flags =
-		    REISERFS_I(inode)->i_flags & ~i_has_xattr_dir;
 	return err;
 }
 
@@ -975,8 +970,6 @@ ssize_t reiserfs_listxattr(struct dentry
 	buf.r_pos = 0;
 	buf.r_inode = dentry->d_inode;
 
-	REISERFS_I(dentry->d_inode)->i_flags |= i_has_xattr_dir;
-
 	err = xattr_readdir(fp, reiserfs_listxattr_filler, &buf);
 	if (err)
 		goto out_dir;
